src/ceph-disk

```diff
@@ -2461,6 +2678,316 @@ def main_activate(args):
 
 ###########################
 
+def _mark_osd_out(cluster, osd_id):
+    LOG.info('Prepare to mark osd.%s out...', osd_id)
+    try:
+        out, ret = command(
+                [
+                    'ceph',
+                    'osd',
+                    'out',
+                    'osd.%s' % osd_id,
+                    ],
+                )
+    except:
+        raise Error('Could not find osd.%s, is a vaild/exist osd id?' % osd_id)
+
+
+def _check_osd_status(cluster, osd_id):
+    """
+    report the osd status:
+    00(0) : means OSD OUT AND DOWN
+    01(1) : means OSD OUT AND UP
+    10(2) : means OSD IN AND DOWN
+    11(3) : means OSD IN AND UP
+    """
+    LOG.info("Checking osd id: %s ..." % osd_id)
+    found = False
+    status_code = 0
+    try:
+        out, ret = command(
+                [
+                    'ceph',
+                    'osd',
+                    'dump',
+                    '--cluster={cluster}'.format(
+                        cluster=cluster,
+                        ),
+                    '--format',
+                    'json',
+                    ],
+                )
+    except subprocess.CalledProcessError as e:
+        raise Error(e)
+    out_json = json.loads(out)
+    for item in out_json[u'osds']:
+        if item.get(u'osd') == int(osd_id):
+            found = True
+            if item.get(u'in') is 1:
+                status_code += 2
+            if item.get(u'up') is 1:
+                status_code += 1
+    if not found:
+        raise Error('Could not osd.%s in osd tree!' % osd_id)
+    return status_code
+
+
+def _remove_osd_directory_files(mounted_path, cluster):
+    """
+    To remove the 'ready', 'active', INIT-specific files.
+    """
+    if os.path.exists(os.path.join(mounted_path, 'ready')):
```

Comment:
hi @dachary,
for the same reason, should we not ignore the activate remove failure?
that both deactivate and activate exist will make sense?
I am not sure, maybe can explain like that deactivate priority is more than activate?